### PR TITLE
Change string formatting check to assertion.

### PR DESCRIPTION
**Pull request by chriseth**
Addresses https://github.com/ethereum/solidity/pull/7770#discussion_r351724479

### CODE REVIEW

File: libdevcore/Exceptions.h [hunk 1]
```diff
@@ -49,6 +49,7 @@ DEV_SIMPLE_EXCEPTION(BadHexCharacter);
 DEV_SIMPLE_EXCEPTION(BadHexCase);
 DEV_SIMPLE_EXCEPTION(FileError);
 DEV_SIMPLE_EXCEPTION(DataTooLong);
```
Comment (axic) [inline libdevcore/Exceptions.h:51]: Why not use `DataTooLong`?

**Comment by axic** [comment]
Why don't we have actual unit tests for this (and not just implicit semantic tests)?

File: libdevcore/Exceptions.h (hunk 1 above)
Comment (chriseth) [reply libdevcore/Exceptions.h:51]: Thought about it, but it's a very different thing.

**Comment by chriseth** [comment]
Unit test for what?

**Review by leonardoalt — APPROVED** [review]

**Comment by axic** [comment]
> Unit test for what?

`dev::formatAsStringOrNumber` in `test/libdevcore` just like we have for `toHex`

**Pull request merged**